### PR TITLE
Master riscv 0.7.1

### PR DESCRIPTION
**Pull request by damonyu1989**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by damonyu1989** [comment]
mkdir build
cd build
cmake .. -DCMAKE_TOOLCHAIN_FILE="../ platforms/linux/riscv64-0.7.1-gcc.toolchain.cmake" -DWITH_OPENCL=OFF -DBUILD_opencv_calib3d=ON -DBUILD_ZLIB=ON -DBUILD_PNG=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_WITH_INSTALL_RPATH=1 -DCMAKE_INSTALL_PREFIX=../install
make

**Comment by damonyu1989** [comment]
The toolchian can be found in https://github.com/damonyu1989/Tools.

**Comment by alalek** [comment]
Do NOT re-open PRs. Update your source branch instead.

refs: #18394 #19768

**Comment by vpisarev** [comment]
@damonyu1989, thank you!
as @alalek said, since the changes are rather local, can you please pull the latest master from opencv repository, make a fresh branch and put your code there?

also, since in OpenCV there is already some RISC-V acceleration, there should be some way to select your version over the other one? Eventually the two implementations should be merged together, but at least they should not conflict with each other.

**Comment by damonyu1989** [comment]
Dear vadim, My code are merged  to the latest master commit. In my opinions,  the PR we pushed, is the version of 0.7.1.  So we add a new file for this version .

**Comment by damonyu1989** [comment]
@vpisarev

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Comment by vpisarev** [comment]
@damonyu1989, I've sent you the version of intrin_rvv071.hpp with some GCC warnings fixed. Please, update it. Also, please, remove trailing whitespaces, which our doc builder complains about, and squash commits into one.

**Review by vpisarev — APPROVED** [review]

**Comment by vpisarev** [comment]
@alalek, the patch is ready to be merged, I think :+1: I tested it my machine under QEMU, works well, all `opencv_test_core` tests pass. Could you please squash the commits and merge it?

File: modules/core/include/opencv2/core/hal/intrin_rvv071.hpp [hunk 1]
```diff
@@ -0,0 +1,2661 @@
...
+//     this list of conditions and the following disclaimer in the documentation
+//     and/or other materials provided with the distribution.
+//
+//   * The name of the copyright holders may not be used to endorse or promote products
+//     derived from this software without specific prior written permission.
+//
+// This software is provided by the copyright holders and contributors "as is" and
+// any express or implied warranties, including, but not limited to, the implied
+// warranties of merchantability and fitness for a particular purpose are disclaimed.
+// In no event shall the Intel Corporation or contributors be liable for any direct,
```
Comment (alalek) [inline modules/core/include/opencv2/core/hal/intrin_rvv071.hpp:36]: Please use short OpenCV license header with "authors" extra comments.

**Comment by vpisarev** [comment]
@damonyu1989, is it fine with you if the license header in intrin_rvv071.hpp is replaced with

```
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html

// Copyright (C) 2015, PingTouGe Semiconductor Co., Ltd., all rights reserved.
```
??

File: modules/core/include/opencv2/core/hal/intrin_rvv071.hpp (hunk 1 above)
Comment (damonyu1989) [reply modules/core/include/opencv2/core/hal/intrin_rvv071.hpp:36]: > @damonyu1989, is it fine with you if the license header in intrin_rvv071.hpp is replaced with
> 
> ```
> // This file is part of OpenCV project.
> // It is subject to the license terms in the LICENSE file found in the top-level directory
> // of this distribution and at http://opencv.org/license.html
> 
> // Copyright (C) 2015, PingTouGe Semiconductor Co., Ltd., all rights reserved.
> ```
> 
> ??
I think it is ok for us.

**Pull request merged**